### PR TITLE
I2C/SPI require device type in constructor

### PR DESCRIPTION
**Pull request by dagar**
In most cases we already have the required Device information at construction, there's no reason to do it later (manual set_device_type() calls). This is another minor simplification to the drivers and a small flash savings.

We're very close to being able to remove CDev from the Device (I2C/SPI) hierarchy entirely.

### CODE REVIEW

**Comment by dagar** [comment]
With the test rack I've confirmed all device ids are consistent with master except for the lsm303d where `DRV_MAG_DEVTYPE_LSM303D` was migrated to `DRV_IMU_DEVTYPE_LSM303D`. The accel and mag on the ST LSM303D are the same physical device.

File: src/drivers/imu/bmi055/BMI055.hpp [hunk 1]
```diff
@@ -53,7 +53,8 @@
 class BMI055 : public device::SPI, public I2CSPIDriver<BMI055>
 {
 public:
-	BMI055(const char *name, const char *devname, I2CSPIBusOption bus_option, int bus, int type, uint32_t device,
+	BMI055(uint8_t devtype, const char *name, const char *devname, I2CSPIBusOption bus_option, int bus, int type,
```
Comment (bkueng) [inline src/drivers/imu/bmi055/BMI055.hpp]: You now have `devtype` and `type`, which are both the same.

File: src/lib/drivers/device/qurt/I2C.cpp [hunk 2]
```diff
@@ -47,17 +47,15 @@
...
 
-I2C::I2C(const char *name, const char *devname, const int bus, const uint16_t address, const uint32_t frequency) :
-	CDev(name, devname),
+I2C::I2C(uint8_t device_type, const char *name, const int bus, const uint16_t address, const uint32_t frequency) :
+	CDev(name, nullptr),
 	_frequency(frequency)
 {
-	DEVICE_DEBUG("I2C::I2C name = %s devname = %s", name, devname);
+	_device_id.devid_s.devtype = device_type;
 	// fill in _device_id fields for a I2C device
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/lib/drivers/device/qurt/I2C.cpp:55]: Done.

File: src/drivers/optical_flow/px4flow/px4flow.cpp [hunk 3]
```diff
@@ -147,11 +145,11 @@ extern "C" __EXPORT int px4flow_main(int argc, char *argv[]);
...
 PX4FLOW::PX4FLOW(I2CSPIBusOption bus_option, int bus, int address, uint8_t sonar_rotation, int bus_frequency,
 		 int conversion_interval, enum Rotation rotation) :
-	I2C("PX4FLOW", PX4FLOW0_DEVICE_PATH, bus, address, bus_frequency),
+	I2C(DRV_FLOW_DEVTYPE_PX4FLOW, MODULE_NAME, bus, address, bus_frequency),
 	I2CSPIDriver(MODULE_NAME, px4::device_bus_to_wq(get_device_id()), bus_option, bus, address),
 	_sonar_rotation(sonar_rotation),
-	_sample_perf(perf_alloc(PC_ELAPSED, "px4f_read")),
-	_comms_errors(perf_alloc(PC_COUNT, "px4f_com_err")),
+	_sample_perf(perf_alloc(PC_ELAPSED, MODULE_NAME": read")),
+	_comms_errors(perf_alloc(PC_COUNT,  MODULE_NAME": com_err")),
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/drivers/optical_flow/px4flow/px4flow.cpp]: Fixed.

File: src/drivers/imu/bmi055/BMI055.hpp (hunk 1 above)
Comment (dagar) [reply src/drivers/imu/bmi055/BMI055.hpp]: Thanks, good eye. Fixed.

**Pull request merged**